js/components/ConsortReport.vue

```diff
@@ -1,14 +1,95 @@
 <template>
-  <div class="container">
-    Hello World From Vue
+  <div id="consort-report-container" class="container" v-if="!loading">
+      <div class="error" v-if="hasError">
+        {{ errorMessage }}
+        <ul v-if="hasErrorDetails">
+          <li v-for="message in errorDetails" :key="message">{{ message }}</li>
+        </ul>
+      </div>
+
+      <ReportCount v-for="report in reportConfig"
+                   :report-name="report.name"
+                   :report-id="report.reportId" />
```

Comment:
If the reports are going to be slow, or resource intensive, or prone to failure, it may make sense to move the logic for fetching the report count into the `ConsortReport` component itself. Arranged that way, you could maintain a queue of pending requests, retry failed requests a limited number of times, and so on. This would also make the `ReportCount` component synchronous, which would simplify testing (dumb components are good).

Also, it may just be me, but I found the name component name `ReportCount` confusing.

Reply:
Would it be clearer to rename it to `ReportSummary`?

Reply:
That sounds reasonable. Here are a couple of the responses I'm getting back when calling `dataService.getReportCount`. I'm not sure if this is Vue causing the issue, or some issue with my REDCap instance.

```
<p class="red" style="max-width:700px;">
	<b>REDCap ERROR:</b> An unknown error has caused the REDCap page to halt. This may have occurred if you have opened
					multiple browser tabs of the same REDCap page. If that is not the case, then please try to reload this page.
</p>
```

and

```
Fatal error: Uncaught exception 'Exception' with message 'Error running External Module query: 
DB Error: Lost connection to MySQL server during query
SQL: SHOW TABLES LIKE 'redcap_external_module%'' in /var/www/html/redcap/external_modules/classes/ExternalModules.php:1213
Stack trace:
#0 /var/www/html/redcap/external_modules/classes/ExternalModules.php(1740): ExternalModules\ExternalModules::query('SHOW TABLES LIK...')
#1 /var/www/html/redcap/external_modules/classes/ExternalModules.php(1702): ExternalModules\ExternalModules::areTablesPresent()
#2 /var/www/html/redcap/external_modules/classes/ExternalModules.php(1610): ExternalModules\ExternalModules::cacheAllEnableData()
#3 /var/www/html/redcap/external_modules/classes/ExternalModules.php(1357): ExternalModules\ExternalModules::getSystemwideEnabledVersions()
#4 /var/www/html/redcap/redcap_v8.6.4/Classes/Hooks.php(37): ExternalModules\ExternalModules::callHook('redcap_every_pa...', Array)
#5 /var/www/html/redcap/redcap_v8.6.4/Classes/System.php(603): Hooks::call('redcap_ in /var/www/html/redcap/external_modules/classes/ExternalModules.php on line 1213
				<div class="red" style="margin:20px 0px;max-width:700px;">
					<b>REDCap crashed due to an unexpected PHP fatal error!</b><br><br>
					<b>Error message:</b> Uncaught exception 'Exception' with message 'Error running External Module query: 
DB Error: Lost connection to MySQL server during query
SQL: SHOW TABLES LIKE 'redcap_external_module%'' in /var/www/html/redcap/external_modules/classes/ExternalModules.php:1213
Stack trace:
#0 /var/www/html/redcap/external_modules/classes/ExternalModules.php(1740): ExternalModules\ExternalModules::query('SHOW TABLES LIK...')
#1 /var/www/html/redcap/external_modules/classes/ExternalModules.php(1702): ExternalModules\ExternalModules::areTablesPresent()
#2 /var/www/html/redcap/external_modules/classes/ExternalModules.php(1610): ExternalModules\ExternalModules::cacheAllEnableData()
#3 /var/www/html/redcap/external_modules/classes/ExternalModules.php(1357): ExternalModules\ExternalModules::getSystemwideEnabledVersions()
#4 /var/www/html/redcap/redcap_v8.6.4/Classes/Hooks.php(37): ExternalModules\ExternalModules::callHook('redcap_every_pa...', Array)
#5 /var/www/html/redcap/redcap_v8.6.4/Classes/System.php(603): Hooks::call('redcap_<br>
					<b>File:</b> /var/www/html/redcap/external_modules/classes/ExternalModules.php<br>
					<b>Line:</b> 1213<br>
				</div>
```

Reply:
@kirenotneb, `ReportSummary` does make more sense, I'll update.